### PR TITLE
angular utilize form validation

### LINKED ISSUE #667: Angular list-form should leverage angular forms
The Angular ListFormComponent should leverage angular forms and their built-in validation system, rather than delegating the validation to listeners of its @Output() field. In this way, the component will only output a message once it gets validated by itself.

### PR DESCRIPTION
**Pull request by Tanya0609**
This PR is to address #667 
Quote from this issue: "The Angular ListFormComponent should leverage angular forms and their built-in validation system, rather than delegating the validation to listeners of its @output() field. In this way, the component will only output a message once it gets validated by itself."

Also, utilized ng-bootstrap library for styling alter. Code Example can be found here.
https://ng-bootstrap.github.io/#/components/alert/examples

**How to test:**
1. generate an angular app with list page selected
2. start the app, add and delete list should work properly.
-  Without enter anything, the Submit button should be disabled
- When enter some chars and completely removed the chars, an error message should show up
ng-form

closes #667

### CODE REVIEW

File: templates/Web/Pages/Angular.List/src/app/app-shell/Param_SourceName_Kebab/list-form/list-form.component.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.col-11{
+    padding-left: 0px;
+    padding-right: 0px;
```
Comment (jsondoo) [inline templates/Web/Pages/Angular.List/src/app/app-shell/Param_SourceName_Kebab/list-form/list-form.component.css:3]: surprised this actaully worked lol nice

**Review by jsondoo — COMMENTED** [review]
Looks good, one small thing though, when I try list with React, I get a warning pressing submit or enter with empty form. For consistency, I guess we should change either this or React to match up?
image

**Comment by SahilTara** [comment]
@jsondoo I would say change react if anything

**Comment by Tanya0609** [comment]
@sahil Agree, we should make changes to React (maybe Vue too if it doesn't do it this way) to handle validation at the form itself.

**Review by jsondoo — DISMISSED** [review]

File: templates/Web/Pages/Angular.List/src/app/app-shell/Param_SourceName_Kebab/list-form/list-form.component.css (hunk 1 above)
Comment (piotrmark) [reply templates/Web/Pages/Angular.List/src/app/app-shell/Param_SourceName_Kebab/list-form/list-form.component.css:3]: image

**Review by piotrmark — APPROVED** [review]

File: templates/Web/Pages/Angular.List/src/app/app-shell/Param_SourceName_Kebab/list.component.html [hunk 2]
```diff
@@ -4,9 +4,7 @@
       <h3>Angular AngularList Template</h3>
     </div>
     <div class="col-12 p-0">
-      <app-list-form
-        (inputText)="handleAddListItem($event)"
-      ></app-list-form>
+      <app-list-form (inputText)="handleAddListItem($event)"></app-list-form>
```
Comment (streetcoder123) [inline templates/Web/Pages/Angular.List/src/app/app-shell/Param_SourceName_Kebab/list.component.html:7]: Is it an intentional change to move it to one line or it is something done automatically by prettier?

File: templates/Web/Projects/AngularDefault/.template.config/template.json [hunk 3]
```diff
@@ -62,7 +62,7 @@
       "manualInstructions": [],
       "actionId": "CB387AC0-16D0-4E07-B41A-F1EA616A7CA9",
       "args": {
-        "dict": "{'@angular/animations': '~7.2.0', '@angular/cdk': '~7.3.7', '@angular/common': '~7.2.0', '@angular/compiler': '~7.2.0', '@angular/core': '~7.2.0', '@angular/forms': '~7.2.0', '@angular/material': '^7.3.7', '@angular/platform-browser': '~7.2.0', '@angular/platform-browser-dynamic': '~7.2.0', '@angular/router': '~7.2.0', 'bootstrap': '^4.3.1', 'core-js': '^2.5.4', 'rxjs': '~6.3.3', 'tslib': '^1.9.0', 'zone.js': '~0.8.26'}",
+        "dict": "{'@angular/animations': '~7.2.0', '@angular/cdk': '~7.3.7', '@angular/common': '~7.2.0', '@angular/compiler': '~7.2.0', '@angular/core': '~7.2.0', '@angular/forms': '~7.2.0', '@angular/material': '^7.3.7', '@angular/platform-browser': '~7.2.0', '@angular/platform-browser-dynamic': '~7.2.0', '@angular/router': '~7.2.0', '@ng-bootstrap/ng-bootstrap': '^4.2.1', 'bootstrap': '^4.3.1', 'core-js': '^2.5.4', 'rxjs': '~6.3.3', 'tslib': '^1.9.0', 'zone.js': '~0.8.26'}",
```
Comment (streetcoder123) [inline templates/Web/Projects/AngularDefault/.template.config/template.json:65]: Just for curiosity, how is the package: `@ng-bootstrap/ng-bootstrap': '^4.2.1'` different from normal `bootstrap` package?

File: templates/Web/Pages/Angular.List/src/app/app-shell/Param_SourceName_Kebab/list.component.html (hunk 2 above)
Comment (Tanya0609) [reply templates/Web/Pages/Angular.List/src/app/app-shell/Param_SourceName_Kebab/list.component.html:7]: automatically by prettier

File: templates/Web/Projects/AngularDefault/.template.config/template.json (hunk 3 above)
Comment (Tanya0609) [reply templates/Web/Projects/AngularDefault/.template.config/template.json:65]: ng-bootstrap is bootstrap specifically for Angular, it is a component based styling. It makes using some Boostrap class with less code. It is highly recommended by the Angular expert and experienced user. More details can be found here. https://ng-bootstrap.github.io/#/components/alert/examples

**Review by streetcoder123 — APPROVED** [review]

**Review by jsondoo — APPROVED** [review]

**Pull request merged**